rust/main/agents/relayer/src/relayer.rs

```diff
@@ -373,40 +277,24 @@ impl BaseAgent for Relayer {
                 application_operation_verifiers.get(destination_domain);
 
             // only iterate through origin chains that were successfully instantiated
-            for (origin, validator_announce) in validator_announces.iter() {
-                let db = match dbs.get(origin) {
-                    Some(db) => db.clone(),
-                    None => {
-                        error!(origin=?origin.name(), "DB missing");
-                        continue;
-                    }
-                };
+            for (origin_domain, origin) in origins.iter() {
+                let db = &origin.database;
+
                 let default_ism_getter = DefaultIsmCache::new(destination_mailbox.clone());
-                let origin_chain_setup = match core.settings.chain_setup(origin) {
```

Comment:
I think we can move `ChainConf` into `Origin` as well. I'll do it for `Destination`.